src/Microsoft.ApplicationInsights/Extensibility/Implementation/Platform/PlatformImplementation.cs

```diff
@@ -44,9 +44,17 @@ public PlatformImplementation()
         /// </summary>
         public string ReadConfigurationXml()
         {
-            // Config file should be in the base directory of the app domain
-            string configFilePath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "ApplicationInsights.config");
-
+            string configFilePath;
+            try
+            {
+                // Config file should be in the base directory of the app domain
+                configFilePath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "ApplicationInsights.config");
+            }
+            catch (SecurityException)
```

Comment:
can Path.Combine throw SecurityException?

Reply:
I don't believe Path.Combine is the culprit here so much as `AppDomain.CurrentDomain.BaseDirectory`

Reply:
right. thanks for clarifying.

Reply:
I dug into this a bit more in the reference source and found that getting the `BaseDirectory` exercises https://referencesource.microsoft.com/#mscorlib/system/AppDomainSetup.cs,056ca0529857f8f8 which is indeed what I'm seeing in my stack traces from Dynamics 

```
Unhandled exception: 
Exception type: System.ServiceModel.FaultException`1[Microsoft.Xrm.Sdk.OrganizationServiceFault]
Message: System.Security.SecurityException: Request for the permission of type 'System.Security.Permissions.FileIOPermission, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089' failed.
   at System.Security.CodeAccessSecurityEngine.Check(Object demand, StackCrawlMark& stackMark, Boolean isPermSet)
   at System.Security.CodeAccessPermission.Demand()
   at System.AppDomainSetup.VerifyDir(String dir, Boolean normalize)
   at Microsoft.ApplicationInsights.Extensibility.Implementation.Platform.PlatformImplementation.ReadConfigurationXml()
   at Microsoft.ApplicationInsights.Extensibility.Implementation.TelemetryConfigurationFactory.Initialize(TelemetryConfiguration configuration, TelemetryModules modules)
   at Microsoft.ApplicationInsights.Extensibility.TelemetryConfiguration.get_Active()
   at LicenseManager.Services.LoggingService..ctor(String loggerName)
   at LicenseManager.UpdateInterceptor..ctor()
```